### PR TITLE
Add PublishEventsProjection and AnalyticsService, #21

### PR DESCRIPTION
**Pull request by patriknw**
References #21

TODO:
- [x] Add test to IntegrationSpec (I have only tried this manually so far)

### CODE REVIEW

File: docs-source/docs/modules/cqrs/examples/shopping-analytics-service-scala/src/main/scala/sample/shoppinganalytics/ShoppingCartEventConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    val consumerSettings =
+      ConsumerSettings(config, new StringDeserializer, new ByteArrayDeserializer)
+        .withBootstrapServers(bootstrapServers)
+        .withGroupId("shopping-cart-analytics")
+        .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    val committerSettings = CommitterSettings(system)
+
+    RestartSource
+      .onFailuresWithBackoff(minBackoff = 1.second, maxBackoff = 30.seconds, randomFactor = 0.1) { () =>
+        Consumer
```
Comment (ennru) [inline docs-source/docs/modules/cqrs/examples/shopping-analytics-service-scala/src/main/scala/sample/shoppinganalytics/ShoppingCartEventConsumer.scala:44]: Create an issue to implement proper shutdown?

File: docs-source/docs/modules/cqrs/examples/shopping-cart-service-scala/src/main/protobuf/ShoppingCartEvents.proto [hunk 2]
```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "sample.shoppingcart.proto";
```
Comment (ennru) [inline docs-source/docs/modules/cqrs/examples/shopping-cart-service-scala/src/main/protobuf/ShoppingCartEvents.proto:4]: Package names by technology are a bit of a smell, it should indicate where these are used. Maybe `events` is better?

**Review by ennru — APPROVED** [review]
Let's take it from here.
LGTM.

File: docs-source/docs/modules/cqrs/examples/shopping-cart-service-scala/src/main/protobuf/ShoppingCartEvents.proto (hunk 2 above)
Comment (patriknw) [reply docs-source/docs/modules/cqrs/examples/shopping-cart-service-scala/src/main/protobuf/ShoppingCartEvents.proto:4]: I placed all grpc and proto generated things there. Often same class names as used elsewhere (ShoppingCart.Event).

**Pull request merged**